data/campaigns/Descent_Into_Darkness/scenarios/12_Endless_Night.cfg

```diff
@@ -68,20 +68,59 @@
 
         [objectives]
             side=1
+            delayed_variable_substitution=yes #so timesForever note works correctly when using debug
             [objective]
                 condition=win
                 description= _ "Defeat the foolish hero"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever less_than_equal_to 1}
+                [/show_if]
             [/objective]
+
+            [objective]
+                condition=win
+                description= _ "Defeat another foolish hero"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever numerical_equals 2}
+                [/show_if]
+            [/objective]
+
+            [objective]
+                condition=win
+                description= _ "Defeat yet another foolish hero"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever greater_than_equal_to 3}
+                [/show_if]
+            [/objective]
+
             [objective]
                 condition=lose
                 description= _ "Destruction of Mal Keshar"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever less_than_equal_to 3}
+                [/show_if]
+            [/objective]
+
+            [objective]
+                {ALTERNATIVE_OBJECTIVE_CAPTION}
+                condition=win
+                description= _ "Destruction of Mal Keshar"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever greater_than_equal_to 4}
+                [/show_if]
             [/objective]
 
             [gold_carryover]
                 carryover_percentage=40
             [/gold_carryover]
 
-            {IS_LAST_SCENARIO}
+            [note]
+                description="<b>" + _"Endless Night" + " " + "$timesForever" + "</b>"
```

Comment:
Why is this split up rather than being a single translatable string? (I guess the formatting doesn't need to be translatable, but I don't see why the number shouldn't be part of the translatable string.)

Reply:
1. It reuses a translated string (it's the same as the scenario title), so it doesn't add to translation work.
2. it was a habit I had gotten into with my own add-ons, to seperate formating from stuff that should be translated vs not be translated.

Reply:
But what if there's a language where it makes more sense to put the number in front of the name rather than after it? I think it might be better to use the version _with_ the variable even for the title.

That said, if you'd rather just leave like this for now, I don't mind; if a translator brings up an issue with it, it can always be fixed then.

Reply:
I agree with @CelticMinstrel. While it's unlikely to be a problem here, it seems to be a poor style.

On the other hand, if you include the number, you're forcing 10 (?) translations.

A more appropriate solution would be to use a string-replacement format such as `_"Endless Night %i"` ... perhaps something is available with WFL?

Reply:
You can use `_"Endless Night $timesForever"` or `_"<b>Endless Night $timesForever</b>"`. Variable substitution happens after translation.

This may still cause plural-form issues, but that's a problem we've always had in WML.

Reply:
I think this probably wouldn't cause plural-form issues, but if it does they can be worked around with Lua.

Reply:
If you really want a subtle counter, then the best way would to show it in the scenario name in objectives; "Endless Night (3)". However, that's not currently possible but the support would probably be easy to add.

Or, perhaps, stricken-through versions of the objective could appear per every defeated foolish hero, so on the third repetition victory objectives would look like this:

~~Defeat the foolish hero~~
~~Defeat the foolish hero~~
Defeat the foolish hero

The list might end up getting too long, though.

Reply:
> ~Defeat the foolish hero~
> ~Defeat the foolish hero~
> Defeat the foolish hero

I really like this idea.

Reply:
> However, that's not currently possible but the support would probably be easy to add.

You can't include variable substitutions inside the title itself, then?

Reply:
@CelticMinstrel Variable substitution doesn't work on keys directly inside [scenario]. It's mostly confined to inside events, with a few exceptions (like [story])

I think the strikethrough would only maybe work with a small, finite number of iterations of the scenario. At some point the rest of the objectives would be pushed off the screen or need a scrollbar. & I think players might find it annoying.